applications/visualizer/frontend/src/components/ViewerSettings.tsx

```diff
@@ -32,14 +37,39 @@ const SyncViewersData = [
   },
 ];
 
-const ViewersList = ["Connectivity graph", "3D viewer", "EM viewer"];
+const viewers = {
+  Graph: twoDViewerWidget(),
+  "3D": threeDViewerWidget(),
+};
 
 const textStyles = { ...secondaryTypographyStyles, fontWeight: 500, flex: 1 };
 const buttonStyle = {
   p: "0.25rem",
 };
 
 const ViewerSettings = ({ open, toggleDrawer }) => {
+  const dispatch = useDispatch();
+  const { workspaces, currentWorkspaceId } = useGlobalContext();
+  const currentWorkspace = workspaces[currentWorkspaceId];
+  const handleToggle = (e, viewer) => {
+    const selectedViewer = viewers[viewer];
+    if (selectedViewer) {
+      let viewerConfig = { ...selectedViewer };
+      if (e.target.checked) {
```

Comment:
suggestion (non-blocking): I would prefer a slightly different approach in the creation of the widget configurations but I think it's a matter of taste:

```
export const threeDViewerWidget = (status = WidgetStatus.ACTIVE) => ({
  id: widgetIds.threeDViewer,
  name: "3D Viewer",
  component: widgetIds.threeDViewer,
  panelName: "rightPanel",
  enableClose: false,
  status,
});

export const twoDViewerWidget = (status = WidgetStatus.ACTIVE) => ({
  id: widgetIds.twoDViewer,
  name: "Connectivity Graph",
  component: widgetIds.twoDViewer,
  panelName: "leftPanel",
  enableClose: false,
  status,
});

const viewers = {
  Graph: twoDViewerWidget,
  "3D": threeDViewerWidget,
};

const handleToggle = (e, viewer) => {
  const status = e.target.checked ? WidgetStatus.ACTIVE : WidgetStatus.MINIMIZED;
  const selectedViewerFunction = viewers[viewer];
  if (selectedViewerFunction) {
    const viewerConfig = selectedViewerFunction(status);
    dispatch(updateWidget(viewerConfig));
    currentWorkspace.changeViewerVisibility(viewer, e.target.checked);
  } else {
    return;
  }
};

```